documentation/start-documenting.rst

```diff
@@ -64,79 +64,109 @@ Building the documentation
+.. _doc-create-venv:
+
+Create a virtual environment
+----------------------------
 
-Using make / make.bat
----------------------
+.. _doc-create-venv-unix:
 
-**On Unix**, run the following from the root of your :ref:`repository clone
-<checkout>` to build the output as HTML::
+**On Unix platforms** that support :program:`make`
+(including Linux, macOS and BSD),
+you can create a new :mod:`venv` with the required dependencies using::
 
-   cd Doc
    make venv
-   make html
 
-or alternatively ``make -C Doc/ venv html``.  ``htmlview`` can be used
-instead of ``html`` to conveniently open the docs in a browser once the
-build completes.
+Building the docs with :program:`make` will automatically use this environment
+without you having to activate it.
+
+.. _doc-create-venv-windows:
+
+**On Windows**, or if not using :program:`make`,
+`create a new virtual environment <venv-create_>`__ manually.
+Always be sure to `activate this environment <venv-activate_>`__
+before building the documentation.
+
+
+.. _building-using-make:
+.. _using-make-make-bat:
+.. _doc-build-make:
+
+Build using make / make.bat
+---------------------------
 
-You can also use ``make help`` to see a list of targets supported by
-:command:`make`.  Note that ``make check`` is automatically run when
-you submit a :ref:`pull request <pullrequest>`, so you should make
-sure that it runs without errors.
+A Unix ``Makefile`` is provided, :cpy-file:`Doc/Makefile`,
+along with a :cpy-file:`Doc/make.bat` batch file for Windows
+that attempts to emulate it as closely as practical.
 
-**On Windows**, the :cpy-file:`Doc/make.bat` batchfile tries to emulate
-:command:`make` as closely as possible, but the venv target is not implemented,
-so you will probably want to make sure you are working in a virtual environment
-before proceeding, otherwise all dependencies will be automatically installed
-on your system.
+.. important::
 
-When ready, run the following from the root of your :ref:`repository clone
-<checkout>` to build the output as HTML::
+   The Windows ``make.bat`` batch file lacks a ``make venv`` target.
+   Instead, it automatically installs any missing dependencies
+   into the currently activated environment (or the base Python, if none).
+   Make sure the environment you :ref:`created above <doc-create-venv-windows>`
+   is `activated <venv-activate_>`__ before running ``make.bat``.
+
+To build the docs as HTML, run::
 
-   cd Doc
    make html
 
-You can also use ``make help`` to see a list of targets supported by
-:cpy-file:`Doc/make.bat`.
+.. tip:: Substitute ``htmlview`` for ``html`` to open the docs in a web browser
+         once the build completes.
+
+To check the docs for common errors with `Sphinx Lint`_
+(which is run on all :ref:`pull requests <pullrequest>`), use::
+
+   make check
+
+To list other supported :program:`make` targets, run::
+
+   make help
+
+See :cpy-file:`Doc/README.rst` for more information.
+
+
+.. _using-sphinx-build:
```

Comment:
This ref target label was added explicitly for backward compatibility, so that existing external URLs (and any implicit internal hyperlinks) that point to this section won't suddenly break. Renaming it would, of course, defeat that whole purpose :)

The up to date ref target label, `doc-build-sphinx`, is immediately below. FWIW, I've adopted the convention of putting the current canonical (i.e. newest) ref target label last, which makes sense both viz the syntax (as the other labels are de facto "aliases" to it), and by the general convention of adding newer things below older ones.